internal/git/gitsync_processor.go

```diff
@@ -168,17 +168,17 @@ func watchRepo(ctx context.Context, r *git.Repository, gitSync *v1.GitSync, rest
 			select {
 			case <-ticker.C:
 				logger.Debug("checking for new updates")
-				patchedResources, recentHash, err := CheckForRepoUpdates(ctx, r, repo, &gitSync.Status, namespace)
+				patchedResources, recentHash, err := CheckForRepoUpdates(ctx, r, repo, lastCommitHash, namespace)
 				if err != nil {
 					return err
 				}
 
-				err = updateCommitStatus(ctx, k8Client, namespacedName, recentHash, repo, logger)
-				if err != nil {
-					return err
-				}
 				// recentHash would be an empty string if there is no update in the  repository
 				if len(recentHash) > 0 {
+					err = updateCommitStatus(ctx, k8Client, namespacedName, recentHash, repo, logger)
+					if err != nil {
+						return err
+					}
```

Comment:
looks good but can you move the call to `ApplyPatchToResources()` above `updateCommitStatus()`? In the case the Controller goes down, we only want to mark this as committed in our Status once it's truly been committed

Reply:
ok done

Reply:
> Also, you need to set `lastCommitHash` to `recentHash`

yeah sorry forgot that